akka-docs-dev/rst/scala/code/docs/stream/GraphStageDocSpec.scala

```diff
@@ -0,0 +1,83 @@
+package docs.stream
+
+import akka.stream.javadsl.Sink
+import akka.stream.scaladsl.Source
+import akka.stream.stage.{ OutHandler, GraphStage, GraphStageLogic }
+import akka.stream._
+
+import akka.stream.testkit.AkkaSpec
+
+import scala.concurrent.{ Await, Future }
+import scala.concurrent.duration._
+
+class GraphStageDocSpec extends AkkaSpec {
+
+  implicit val mat = ActorMaterializer()
+
+  "Demonstrate creation of GraphStage boilerplate" in {
+    //#boilerplate-example
+    import akka.stream.SourceShape
+    import akka.stream.stage.GraphStage
+
+    class NumbersSource extends GraphStage[SourceShape[Int]] {
+      // Define the (sole) output port of this stage
+      val out: Outlet[Int] = Outlet("NumbersSource")
+      // Define the shape of this stage, which is SourceShape with the port we defined above
+      override val shape: SourceShape[Int] = SourceShape(out)
+
+      // This is where the actual (possibly stateful) logic will live
+      override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = ???
+    }
+    //#boilerplate-example
```

Comment:
maybe, but I wanted to show the basic GraphStage construction since this generalizes to the others. We will need more complex examples for sure.